GTG/core/timer.py

```diff
@@ -82,3 +86,35 @@ def set_configuration(self, refresh_hour, refresh_min):
 
     def get_configuration(self):
         return self.config.get('hour'), self.config.get('min')
+
+    def time_parser(self, time):
+        hour_value = ''
+        minute_value = ''
+        # For clarity, I'm using two match criteria:
+        # Format for HH:MM and optional am/pm
+        match_1 = re.match('([0-9]{1,2})(:)([0-9]{2})( )?([ap]m)?', time)
+        # Format for HH and am/pm
+        match_2 = re.match('([0-9]{1,2})( )?([ap]m)', time)
+        if match_1:
+            hour_value = match_1.group(1)
+            minute_value = match_1.group(3)
+            if match_1.group(5) == 'pm':
+                hour_value = str(int(match_1.group(1)) + 12)
+            correct_hour = int(hour_value) >= 0 and int(hour_value) < 24
+
+        elif match_2:
+            minute_value = '00'
+            if match_2.group(3) == 'am':
+                hour_value = match_2.group(1)
+                correct_hour = int(hour_value) >= 0 and int(hour_value) < 13
+
+            elif match_2.group(3) == 'pm':
+                hour_value = str(int(match_2.group(1)) + 12)
+                correct_hour = int(hour_value) > 12 \
+                    and int(hour_value) < 24
+
+        correct_min = int(minute_value) >= 0 and int(minute_value) < 60
```

Comment:
thanks, I'm applying this to all occurences of <= / > ...

hmm, why do the datetime.time validation twice? I mean, why not keep parse_time() purely to do the parsing? My parse_time() is good enough I think to analyse and deduce what is and what is not allowed. time_changed() takes care of validating if the time is really what it what it should be, a.k.a. uses datetime.time. 

I think this is a good procedure, but I might be missing a point. Keeping this as it was, though for now